spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+[source,java]
+----
+@Outgoing("my-channel")                                        // <1>
+public Message<String> produce() {                             // <2>
+  return Message.of("hello");                                  // <3>
+}
+----
+1. `my-channel` is the targeted channel
+2. the method is called for every consumer _request_
+3. you can create a _plain_ `org.eclipse.microprofile.reactive.messaging.Message` using `org.eclipse.microprofile.reactive.messaging.Message#of(T)`
+
+Reactive Messaging support various form of method signatures.
+This is detailed in the next section.
+
+A method annotated with `@Outgoing` is a Reactive Streams _publisher_ and so publishes messages according to the request it receives.
+
+===== Method consuming and producing
+
+A method can combine the `@Incoming` and `@Outgoing` annotation and acts as a Reactive Streams processor:
+
+[source,java]
+----
+@Incoming("my-incoming-channel")                            // <1>
+@Outgoing("my-outgoing-channel")                            // <2>
+public Message<String> process(Message<String> message) { 
+  return Message.of(message.getPayload().toUpperCase());
+}
+----
+1. The incoming channel
+2. The outgoing channel
+
+==== Connectors and Messaging Providers
+
+The application can receive and forward messages from various message broker or transport layer.
+For instance, an application can be connected to a Kafka cluster, an AMQP broker or an MQTT server.
 
-An implementation is expected to advertise and allow the differentiation between different messaging providers by providing subclasses of `MessagingProvider`. For example, a Kafka messaging provider may be provided by a class that looks like this:
+Reactive Messaging _Connectors_ are extension managing the communication with a specific transport technology.
+They are responsible for mapping specific _channel_ to remote sink or source of messages.
+This mapping is configured in the application configuration.
+Note that implementation may provide various way to configure the mapping, but the support of MicroProfile Config is mandatory.
+
+Connectors are associated with a `org.eclipse.microprofile.reactive.messaging.MessagingProvider` class corresponding to a messaging transport, such as Apache Kafka, Amazon Kinesis, RabbitMQ or Apache ActiveMQ.
```

Comment:
Note that an implementation may provide various ways to configure the mapping, but support for MicroProfile Config as a configuration source is mandatory.